### PR TITLE
feat: merge payment-mvp branch into develop

### PR DESCRIPTION
**Pull request by wanlingt**
## Solution

This PR merges the branch used for the current payments MVP into develop. Users can use payments if they have the beta flag enabled.

**Breaking Changes** 

- [x] No - this PR is backwards compatible  

## Tests

Tests below are taken from the beta flag PR #5831

On an admin without payment beta flag:
- [x] Go to the settings tab of any form. Ensure that payments is not displayed
- [x] Go to the create tab of any form. Ensure that payments tab in the side bar is not displayed
- [x] Try calling the API for stripe Oauth at `/stripe`. Ensure that you receive an error
- [x] Try calling the API for payment update at `/payment`. Ensure that you receive an error

On an admin with payment beta flag (or add the payment flag):
- [x] Set up a storage mode form. Ensure you can connect your stripe account
- [x] Update payment fields
- [x] Open the form and make a successful payment. Ensure the results are logged in the result tab.

## Deploy Notes

**New environment variables**:

- `PAYMENT_DEFAULT_CURRENCY` : Default currency for all payments
- `PAYMENT_STRIPE_PUBLISHABLE_KEY` : Stripe Account Publishable Key
- `PAYMENT_STRIPE_SECRET_KEY` : Stripe account Secret Key
- `PAYMENT_STRIPE_CLIENT_ID` : Stripe client ID
- `PAYMENT_STRIPE_WEBHOOK_SECRET`: Stripe webhook secret

**New scripts**:

- `add-payment-flag.js` : mongo db calls to update payment beta flag for specified users

**New dependencies**:

- `stripe/react-stripe-js`
- `stripe/stripe-js`
- `stripe-event-types`
- `stripe`

### CODE REVIEW

File: src/app/loaders/express/session.ts [hunk 1]
```diff
@@ -20,7 +20,7 @@
   })
 
   return [
-    cookieParser(), // CookieParser should be above session
+    cookieParser(config.sessionSecret), // CookieParser should be above session
```
[the remark this answers is not in the document]
Comment (wanlingt) [reply src/app/loaders/express/session.ts:23]: these are local testing keys and can't be used in a production environment

cc @justynoh

**Comment by wanlingt** [comment]
> #### ⚠️ GitGuardian has uncovered 4 secrets following the scan of your pull request.
> Please consider investigating the findings and remediating the incidents. Failure to do so may lead to compromising the associated services or software components.
> 
> 🔎 Detected hardcoded secrets in your pull request
> 
> GitGuardian id	Secret	Commit	Filename	
> [5053850](https://dashboard.gitguardian.com/incidents/5053850?occurrence=70321543)	Stripe Webhook Secret	[425ff79](https://github.com/opengovsg/FormSG/commit/425ff79fd765d746e1b35754bf6e8109e62af4cf)	src/app/config/features/payment.config.ts	[View secret](https://github.com/opengovsg/FormSG/commit/425ff79fd765d746e1b35754bf6e8109e62af4cf#diff-b71d8feb08d17793443dfbce6bc9df3c9dc9a98570d68c84d00d386aed95c92fR47)
> [5053850](https://dashboard.gitguardian.com/incidents/5053850?occurrence=70322324)	Stripe Webhook Secret	[2d77896](https://github.com/opengovsg/FormSG/commit/2d778964a80e0f93798f1f809ea056bf6a429aae)	src/app/config/features/payment.config.ts	[View secret](https://github.com/opengovsg/FormSG/commit/2d778964a80e0f93798f1f809ea056bf6a429aae#diff-b71d8feb08d17793443dfbce6bc9df3c9dc9a98570d68c84d00d386aed95c92fL47)
> [5053850](https://dashboard.gitguardian.com/incidents/5053850?occurrence=70322325)	Stripe Webhook Secret	[afcf0ca](https://github.com/opengovsg/FormSG/commit/afcf0caa1f3e30d870d1b92fd196a79bca34a043)	src/app/config/features/payment.config.ts	[View secret](https://github.com/opengovsg/FormSG/commit/afcf0caa1f3e30d870d1b92fd196a79bca34a043#diff-b71d8feb08d17793443dfbce6bc9df3c9dc9a98570d68c84d00d386aed95c92fR47)
> [5053850](https://dashboard.gitguardian.com/incidents/5053850?occurrence=70322326)	Stripe Webhook Secret	[74a84ad](https://github.com/opengovsg/FormSG/commit/74a84add27e5a394e2d88f270e4734006f5b8182)	src/app/config/features/payment.config.ts	[View secret](https://github.com/opengovsg/FormSG/commit/74a84add27e5a394e2d88f270e4734006f5b8182#diff-b71d8feb08d17793443dfbce6bc9df3c9dc9a98570d68c84d00d386aed95c92fL47)
> 🛠 Guidelines to remediate hardcoded secrets
> 
> 1. Understand the implications of revoking this secret by investigating where it is used in your code.
> 2. Replace and store your secrets safely. [Learn here](https://blog.gitguardian.com/secrets-api-management?utm_source=product&utm_medium=GitHub_checks&utm_campaign=check_run_comment) the best practices.
> 3. Revoke and [rotate these secrets](https://docs.gitguardian.com/secrets-detection/detectors/specifics/stripe_webhook_secret#revoke-the-secret?utm_source=product&utm_medium=GitHub_checks&utm_campaign=check_run_comment).
> 4. If possible, [rewrite git history](https://blog.gitguardian.com/rewriting-git-history-cheatsheet?utm_source=product&utm_medium=GitHub_checks&utm_campaign=check_run_comment). Rewriting git history is not a trivial act. You might completely break other contributing developers' workflow and you risk accidentally deleting legitimate data.
> 
> To avoid such incidents in the future consider
> 
> * following these [best practices](https://blog.gitguardian.com/secrets-api-management/?utm_source=product&utm_medium=GitHub_checks&utm_campaign=check_run_comment) for managing and storing secrets including API keys and other credentials
> * install [secret detection on pre-commit](https://https://docs.gitguardian.com/ggshield-docs/integrations/git-hooks/pre-commit?utm_source=product&utm_medium=GitHub_checks&utm_campaign=check_run_comment) to catch secret before it leaves your machine and ease remediation.
> 
> 🦉 [GitGuardian](https://dashboard.gitguardian.com/auth/login/?utm_medium=checkruns&utm_source=github&utm_campaign=cr1) detects secrets in your source code to help developers and security teams secure the modern development process. You are seeing this because you or someone else with access to this repository has authorized GitGuardian to scan your pull request.Our GitHub checks need improvements? [Share your feedbacks](https://050127dayho.typeform.com/to/KmeAPTMk)!

these were local testing keys that can't be used in a production environment

cc @justynoh

**Review by timotheeg — APPROVED** [review]

**Comment by justynoh** [comment]
^ yep, keys are specifically associated with webhook receiving on my local machine only

File: src/app/loaders/express/session.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (timotheeg) [reply src/app/loaders/express/session.ts:23]: @wanlingt I'm going to dismiss this for now, but I think it'll be worth doing some code review/audit and satisfy ourselves that our session system is OK. I'll open a ticket for it.

**Pull request merged**